Update README.md (#1727)
smartdevicelink/sdl_ios: README.md

### Installing
To install this library as a framework in your app, see the [Installation Guide](https://smartdevicelink.com/en/guides/iOS/getting-started/installation/).

#### Adding a Dynamic Framework
Tagged to our releases is a dynamic framework file that can be drag-and-dropped into the application. Dynamic frameworks are supported on iOS 8+. **WARNING: You cannot submit your app to the app store with the framework as is. You MUST strip the simulator part of the framework first. Strip the x64 and i386 portions first like so:**

```bash
lipo -remove i386 -remove x86_64 -o SmartDeviceLink.framework/SmartDeviceLink SmartDeviceLink.framework/SmartDeviceLink`
```

You can check the current architectures like so:

```bash
lipo -info SmartDeviceLink.framework/SmartDeviceLink
```

### Getting Help
If you have questions, first view our guides on the [SmartDeviceLink website](https://smartdevicelink.com/en/guides/iOS/getting-started/installation/).

> If you see a bug, please post an issue on the appropriate repository. Please see the [contribution guidelines](https://github.com/smartdevicelink/sdl_ios/blob/master/CONTRIBUTING.md) before proceeding. If you need general assistance, or have other questions, you can [sign up](http://slack.smartdevicelink.org) for the [SDL Slack](https://smartdevicelink.slack.com) and chat with other developers and the maintainers of the project.

Author: Updated a modified version. The sign up link was incorrect and the change for the slack link was correct originally.

### Example Apps

> If you'd like to use the Hello SDL example apps and have Cocoapods installed, you can use `pod try SmartDeviceLink`, alternately, you can clone or download the project.

Reviewer: ```suggestion
The library contains two example apps: one written in Objective-C and one in Swift. If you have CocoaPods installed, you can easily run one of the example apps by executing `pod try SmartDeviceLink` in your terminal. Alternatively, you can clone or download the project, but before attempting to build and run the example apps you must follow the [installation steps](#preparing-for-development).
```

Author: Updated but modified slightly

> If you'd like to use the Hello SDL example apps and have Cocoapods installed, you can use `pod try SmartDeviceLink`, alternately, you can clone or download the project.

Author: Yeah, I wasn't sure what the call them.

> There are additional example apps available [in the example organization](https://github.com/smartdevicelink-examples/), these require Cocoapods to install dependencies.

Reviewer: ```suggestion
There are additional example apps available [in the example organization](https://github.com/smartdevicelink-examples/), these require CocoaPods to install dependencies.
```

3. We welcome PRs for bug fixes! If there's a confirmed issue for a bug, we would welcome a bug fix. The PR will require a review from a project maintainer before it can be merged.
4. PRs for features require additional planning. Features must be approved by the SDLC using the [Evolution Process](https://github.com/smartdevicelink/sdl_evolution) before an issue can be opened. Furthermore, a feature must be approved to be developed for a release by the SDLC before a PR can be opened. Finally, a PR cannot be merged if the underlying feature requires or should have alignment across multiple projects until PRs for all those projects are ready for review.

#### Preparing for Development
To prepare the library for development, you will need to take a few steps:

1. Clone this repository.
1. Install submodules with `git submodule init` and `git submodule update`.
1. Install [testing dependencies](#running-tests).

> 1. Ensure that you can run the [Hello SDL test app](#example-apps).

Reviewer: ```suggestion
1. Ensure that you can run the [Example test apps](#example-apps).
```

At this point, you can run tests from Xcode, or, if you wish to run the tests exactly as they will be run on the CI server, run:

```bash
xcodebuild build-for-testing -project SmartDeviceLink-iOS.xcodeproj -destination platform=iOS Simulator,name=iPhone 11,OS=13.5 -scheme SmartDeviceLink

set -o pipefail && xcodebuild test-without-building -project SmartDeviceLink-iOS.xcodeproj -destination platform=iOS Simulator,name=iPhone 11,OS=13.5 -scheme SmartDeviceLink -configuration Debug ONLY_ACTIVE_ARCH=NO RUN_CLANG_STATIC_ANALYZER=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ENABLE_TESTABILITY=YES
```

##### Lock Screen Screenshot Tests

> We run some additional tests using [FBSnapshotTestCase](https://github.com/uber/ios-snapshot-test-case). These tests generate the lock screen view controller and compare it to generated screenshots. By default, the generated screenshots use the iPhone 11 simulator. If you run unit tests against those simulators, the tests should pass.

Author: Updated slightly modified
